text/00110-cli-framework-compatibility-strategy.md

```diff
@@ -0,0 +1,464 @@
+and if its optional, the CLI has to, at compile time, treat the `undefined` scenario, hence, supporting earlier schemas that didn't have this property.
+
+Removing this mechanism will mean that instead of, theoretically, having only place consider older versions of the schema,
+we will now have multiple places in the code that should be aware of it. But thats ok, because its enforced by the compiler,
+and it also provides better context and granularity for doing that so called evolution.
+
+Also, in reality, we don't actually do any evolution. All we do is synthetically upgrade the schema version,
+without adding or modifying any properties.
+
+This brings us to the next item.
+
+##### Compatibility Validation
+
+This is done in the [`verifyManifestVersion`](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/cx-api/lib/versioning.ts#L39) function,
+and contains two validations:
+
+1. **Framework cannot be bigger than CLI.**
+
+   We have already [described](#step-3-validate-cli--framework) how to handle this validation.
+
+2. **CLI cannot be bigger than Framework.**
+
+   This validation is actually misleading, since we do support CLI > Framework. The way its currently implemented is by doing
+   the aforementioned evolution. But, if the evolution is not necessary anymore, than this validation is not necessary anymore.
+   Which makes sense, since we do support this scenario.
+
+### CLI Regression Tests
+
+The second major mechanism we need to implement is CLI regression tests.
+
+As already mentioned, `cx-protocol` API breakage is not the only thing that can go wrong.
+This is to ensure that we don't introduce breaking changes to the CLI itself. The plan here is fairly straight forward:
+
+```console
+git checkout v${LATEST_PUBLISHED_VERSION}
+npm install /path/to/aws-cdk
+npm install /path/to/packages/*
+cd packages/aws-cdk
+yarn integ-cli
+```
+
+> Not to be taken literally, commands just illustrate intent
+
+# Drawbacks
+
+The only drawback I can think of is that we will be forcing users to upgrade their CLI when they upgrade their framework.
+This is an additional action that they didn't have to take until now. However, calling this a *drawback* is debatable,
+since it actually does provide a more consistent behavior than we have now. In any case, we don't consider this a sufficiently serious issue.
+
+# Rationale and Alternatives
+
+## New Compatibility Model
+
+As we realized by now, this RFC proposes a significant change to our compatibility model.
+Instead of attempting to have new framework versions support old CLI versions,
+we will now require that CLI `>=` Framework. Forcing users to upgrade their CLI every time they upgrade their framework.
+
+### Why is it OK to enforce this requirement?
+
+- We want to migrate most of the functionality to the framework. Making CLI upgrades especially safe.
+- Users tend to upgrade the CLI before upgrading the framework.
```

Comment:
`[citation needed]`

Reply:
Well this is more of an educated guess. I'll rephrase and add some context.